examples_tests/48.ArithmeticUnitTest/main.cpp

```diff
@@ -0,0 +1,230 @@
+#define _IRR_STATIC_LIB_
+#include <irrlicht.h>
+
+#include "../../source/Irrlicht/COpenGLDriver.h"
+
+
+using namespace irr;
+using namespace core;
+using namespace video;
+using namespace asset;
+
+enum TestOperation {
+	TO_AND = 1,
+	TO_XOR,
+	TO_OR,
+	TO_ADD,
+	TO_MUL,
+	TO_MIN,
+	TO_MAX
+};
+
+template<typename T, typename OP, uint32_t subgroupSize>
+T emulatedSubgroupReduction(const T* data, const uint32_t globalInvocationIndex)
+{
+	auto subgroupID = globalInvocationIndex / subgroupSize;
+	//auto subgroupOffset = data + subgroupID * subgroupSize;
```

Comment:
this formula is correct, although faster thing would be to `assert` that "subgroupSize is power of two"

then compute `subgroupOffset = data+localInvocationIndex&(subgroupSize-1u);`

Reply:
well, that was the code you provided me with, and `subgroupOffset` was never used

Reply:
it was to emulate a single workgroup, not the whole dispatch